go/vt/vtgate/executor.go

```diff
@@ -114,16 +116,17 @@ const pathScatterStats = "/debug/scatter_stats"
 const pathVSchema = "/debug/vschema"
 
 // NewExecutor creates a new Executor.
-func NewExecutor(ctx context.Context, serv srvtopo.Server, cell string, resolver *Resolver, normalize bool, streamSize int, cacheCfg *cache.Config) *Executor {
+func NewExecutor(ctx context.Context, serv srvtopo.Server, cell string, resolver *Resolver, normalize, warnOnShardedOnly bool, streamSize int, cacheCfg *cache.Config) *Executor {
 	e := &Executor{
```

Comment:
we can club (warnOnShardedOnly bool, streamSize int, cacheCfg *cache.Config) the config and send a singe executor init config, it will be easier to add new config as well.

Reply:
On the other hand, we do not need to pass this around as this is available in the same package.